### ISSUE TITLE
Automatic Concentration doesn't always work

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#545]
In GitLab by @Runesmite on Sep 26, 2021, 16:37

I am using Midi-QOL and CUB. I have CUB's concentration automation turned off.

When spells are cast certain spells will create the effect of "concentrating" while other spells do not. I have deleted and re-added spells and it happens the same.

For example, Bless and Pass without trace work fine, but Stinking Cloud and Hunter's Mark do not.

I'm not sure why they are not working, but I would be happy to provide any files you need.

Thank you.

Edit:
Sorry, I was tired when I first put this in! Other useful information:
- Foundry Edition: **0.8.9**
- DnD5e - Fifth Edition System: **1.4.3**
- Midi QOL: **0.8.61**

**Comment by tposney** [comment]
Out of interest which hunter's mark are you using?
I believe the cause of the problem is the well know "Race condition" that get's talked about in found pre 0.9. In effect two effects are applied (hunter's mark it is the concentration and bonus), stinking cloud I'm not quite sure about, and there is a raced to see which or both get applied. 

I've spent a bit of time testing this in 0.9 of foundry and it seems to be fixed.

In any event can you attack the two items that are a problem plus your midi-qol settings to the report (midi-qol settings can be exported from the misc tab).

**Comment by tposney** [comment]
In GitLab by @Runesmite on Sep 27, 2021, 24:21

I have used the standard Hunter's Mark from the SRD spells compendium as well as a different imported version. I've tried changing the targeting options of each spell as well. 

I did try re-adding Stinking cloud from the SRD compendium and that one does work now with Automatic Concentration! So, it seems now the only spell not working is Hunter's Mark, any variation on our Ranger. 

I've attached the Midi QOL settings [fvtt-midi-qol-settings.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/5e87d2c3741403f94664b085d56e668bec3820d2c9581729a9ca0321b591ac76/fvtt-midi-qol-settings.json)

I wasn't sure how you wanted the spells attached so I did a screenshot ![Concentration_Issue](https://my-gitlab-bucket.s3.amazonaws.com/640711674/8bc9ad79962e9e5631915bc1f211baf1f33768d7a33cf245308729294a6fda1d/Concentration_Issue.png)

**Comment by tposney** [comment]
In GitLab by @Runesmite on Sep 27, 2021, 24:36

I did even more testing, I found that if I remove the "Utility" damage roll from Hunter's Mark, and don't set it at all, the concentration will work as designed. However, if the "Damage" box appears on the card at all, Concentration will not apply for Hunter's Mark.

**Comment by tposney** [comment]
I think I know what is happening. If an item as a damage roll midi considers the entire use of the item 'incomplete' until the damage is rolled, it can't tell if the damage roll is for information or is actually meant to damage. 

Concentration is applied after the roll is complete, since it also records which targets were hit/did not save so it can remove effects when concentration is removed and also to avoid applying concentration if you did not hit anyone/everyone saved etc. - which is useful as it streamlines the concentration case when the spell "did not work".

I tested this by clicking the damage button for Hunter's mark and concentration was added.

So either roll the damage or remove the damage from item
